k8s/test-deployment.py

```
@@ -0,0 +1,437 @@
+                        matching_services = [s for s in services.values()
+                                           if s['name'] == service_name and s['namespace'] == ingress['namespace']]
+
+                        if not matching_services:
+                            self.test_results['failed'].append(f"Ingress {ingress['name']} references non-existent service {service_name}")
+                        else:
+                            self.test_results['passed'].append(f"Ingress {ingress['name']} references valid service {service_name}")
+
+    def _test_rbac_consistency(self, resources: Dict[str, Any]):
+        """Test RBAC consistency"""
+        role_bindings = {k: v for k, v in resources.items() if v['kind'] == 'RoleBinding'}
+        cluster_role_bindings = {k: v for k, v in resources.items() if v['kind'] == 'ClusterRoleBinding'}
```

Ruff (0.12.2)

Local variable `cluster_role_bindings` is assigned to but never used

Remove assignment to unused variable `cluster_role_bindings`

(F841)